### ISSUE TITLE
Remove ensemble dimension in loss

### ISSUE THREAD

**Issue by VeraChristina** [ecmwf/anemoi-core#757]
### What maintenance does this project need?

Probabilistic losses do not contain an ensemble dimension as the target is the deterministic analysis. Therefore it is not needed either in probabilistic or deterministic losses. @mc4117 suggested that, for clarity, we should consider removing the ensemble dimension in the loss.

The refactor would touch different parts of the training code, including the plotting callbacks, where this would lead to more clarity and simplify the existing code.

**Comment by frazane** [comment]
Superseeded by https://github.com/ecmwf/anemoi-core/pull/1154?
